### PR TITLE
gui: Beautify odometer screen

### PR DESCRIPTION
**Pull request by MistrKernnunos**
BFW-2113

### CODE REVIEW

File: src/gui/screen_menu_odometer.cpp [hunk 1]
```diff
@@ -45,32 +26,17 @@ class ScreenMenuOdometer : public AddSuperWindow<screen_t> {
     }
 
 public:
-    ScreenMenuOdometer();
+    ScreenOdometer()
+        : OdometerScreen(_(label)) {
+        Odometer_s::instance().force_to_eeprom();
+        Item<1>().UpdateValue(Odometer_s::instance().get_from_eeprom(Odometer_s::axis_t::X));
```
Comment (RadekVana) [inline src/gui/screen_menu_odometer.cpp:32]: index with type, would be more readable

File: src/guiapi/include/WindowItemFormatableLabel.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    const std::function<void(char *)> printAs;
+
+protected:
+    ValueType value;
+    ValueType oldVal;
+    constexpr static const char *NA = N_("N/A");
+    constexpr static const char *NI = N_("Not initialized");
+
+    void printExtension(Rect16 extension_rect, color_t color_text, color_t color_back, ropfn raster_op) const override {
```
Comment (RadekVana) [inline src/guiapi/include/WindowItemFormatableLabel.hpp:28]: add non-template parent containing this method (and everything else not depended on ValueType)

File: src/guiapi/src/MItem_tools.cpp [hunk 3]
```diff
@@ -500,3 +501,45 @@ void MI_FS_AUTOLOAD::OnChange(size_t old_index) {
...
+    : MI_ODOMETER_DIST(_(label), 0, is_enabled_t::yes, is_hidden_t::no, -1) {
+}
+MI_ODOMETER_TIME::MI_ODOMETER_TIME()
+    : WI_FORMATABLE_LABEL_t<uint32_t>(_(label), 0, is_enabled_t::yes, is_hidden_t::no, 0, [&](char *buffer) {
+        time_t time = (time_t)value;
+        constexpr static uint32_t secPerDay = 24 * 60 * 60;
+        constexpr static uint32_t secPerHour = 60 * 60;
+        const struct tm *timeinfo = localtime(&time);
+        if (timeinfo->tm_yday) {
+            uint16_t days = value / secPerDay;
```
Comment (RadekVana) [inline src/guiapi/src/MItem_tools.cpp:534]: please comment why did you recalculate days and hours and did not use values from `timeinfo`

**Review by RadekVana — CHANGES_REQUESTED** [review]

File: src/guiapi/include/WindowItemFormatableLabel.hpp (hunk 2 above)
Comment (MistrKernnunos) [reply src/guiapi/include/WindowItemFormatableLabel.hpp:21]: 👍

File: src/guiapi/src/MItem_tools.cpp (hunk 3 above)
Comment (MistrKernnunos) [reply src/guiapi/src/MItem_tools.cpp:534]: 👍

**Review by RadekVana — APPROVED** [review]

**Pull request merged**